src/main/java/PersonalBudget/business/user/domain/service/UserService.java

```diff
@@ -32,6 +32,8 @@ public boolean isEmail(UserDTO userDTO) {
     public void addNewUser(UserDTO userDTO) {
         UserEntity userEntity = userMapper.mapUserDTOToUserEntity(userDTO, bCryptPasswordEncoder);
         userRepository.save(userEntity);
```

Comment:
save() returns the saved entity so you can probably retrieve the id from the returned object.